hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineMetadataUtils.java

```diff
@@ -181,4 +140,17 @@ public static <T extends SpecificRecordBase> T deserializeAvroMetadata(InputStre
       return fileReader.next();
     }
   }
+
+  public static <T extends SpecificRecordBase> Option<HoodieInstantWriter> getInstantWriter(Option<T> metadata) {
+    if (metadata.isEmpty()) {
+      return Option.empty();
+    }
+    return Option.of(outputStream -> {
+      DatumWriter<T> datumWriter = (DatumWriter<T>) DATUM_WRITERS.computeIfAbsent(metadata.get().getClass(), SpecificDatumWriter::new);
```

Comment:
Is `SpecificDatumWriter` thread-safe, i.e., using the same `SpecificDatumWriter` instance to serialize metadata concurrently does not incur any correctness issue?